libs/sdk-backend-spi/src/workspace/dashboards/index.ts

```diff
@@ -481,3 +489,59 @@ export interface IWorkspaceDashboardsService {
      */
     validateDashboardsExistence(dashboardRefs: ObjRef[]): Promise<IExistingDashboard[]>;
 }
+
+/**
+ * Service to query dashboards.
+ *
+ * @public
+ */
+export interface IDashboardsQuery {
+    /**
+     * Sets number of dashboards to return per page.
+     * Default size: 50
+     *
+     * @param size - desired max number of dashboards per page must be a positive number
+     * @returns dashboards query
+     */
+    withSize(size: number): IDashboardsQuery;
+
+    /**
+     * Sets starting page for the query. Backend WILL return no data if the page is greater than
+     * total number of pages.
+     * Default page: 0
+     *
+     * @param page - zero indexed, must be non-negative
+     * @returns dashboards query
+     */
+    withPage(page: number): IDashboardsQuery;
+
+    /**
+     * Sets filter for the query.
+     *
+     * @param filter - Filtering parameter in RSQL
+     * @returns dashboards query
+     */
+    withFilter(filter: string): IDashboardsQuery;
```

Comment:
I don't think leak RSQL outside is a good idea... And what about if we add more filtering options.